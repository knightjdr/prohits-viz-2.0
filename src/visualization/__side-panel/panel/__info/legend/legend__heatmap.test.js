import { shallow } from 'enzyme';

import HeatmapLegend from './legend__heatmap';

const params = {
  abundanceCap: 50,
  abundanceName: 'Abundance',
  gradientFill: ['#ffffff', '#ff0000', '#000000'],
  minAbundance: 0,
};

describe('Heatmap legend', () => {
  it('should render', () => {
    const wrapper = shallow(
      HeatmapLegend({ ...params }),
    );
    expect(wrapper).toMatchSnapshot();
  });
});
