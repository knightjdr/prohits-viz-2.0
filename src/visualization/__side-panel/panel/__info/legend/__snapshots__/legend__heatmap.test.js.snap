// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heatmap legend should render 1`] = `
<svg
  height="80"
  id="legend"
  viewBox="0 0 200 80"
  width="200"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <linearGradient
      id="legendGradient"
    >
      <stop
        offset="0%"
        stopColor="#ffffff"
      />
      <stop
        offset="50%"
        stopColor="#ff0000"
      />
      <stop
        offset="100%"
        stopColor="#000000"
      />
    </linearGradient>
  </defs>
  <g>
    <text
      textAnchor="middle"
      x="100"
      y="20"
    >
      Abundance
    </text>
    <rect
      fill="url('#legendGradient')"
      height="20"
      width="150"
      x="25"
      y="30"
    />
    <text
      textAnchor="middle"
      x="25"
      y="65"
    >
      0
    </text>
    <text
      textAnchor="middle"
      x="175"
      y="65"
    >
      50
    </text>
  </g>
</svg>
`;
